### PR TITLE
Draw manager

### PR DESCRIPTION
**Pull request by LeviSchuck**
Refactor the screenshot get-canvas code into a draw manager, rather than being in the overlay class which is called in Hooks.

This may allow for mixins to register and unregister interest in the canvas too.

It was originally coupled with plugins, but Adam recommended to use a sort of listener register/unregister.

This will make it so I can sleep at the right time for another PR ( #1350 ) which will do frame limiting.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/ui/DrawListener.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui;
+
+import java.awt.image.BufferedImage;
+
+public interface DrawListener
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ui/DrawListener.java]: This entire interface should probably just be `Consumer<BufferedImage>`

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotDrawListener.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.awt.image.BufferedImage;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.function.Consumer;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.client.ui.DrawListener;
+
+@Slf4j
+public class ScreenshotDrawListener implements DrawListener
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotDrawListener.java]: This whole class seems to be duplicated by draw manager now. Why not just use it?

File: runelite-client/src/main/java/net/runelite/client/ui/DrawListener.java (hunk 1 above)
Comment (LeviSchuck) [reply runelite-client/src/main/java/net/runelite/client/ui/DrawListener.java]: Sure thing.

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotDrawListener.java (hunk 2 above)
Comment (LeviSchuck) [reply runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotDrawListener.java]: A list of consumers to provide buffered images is not the same as a queue of consumers to process for a specific plugin’s use.

**Pull request merged**